src/ngraph_builder.cc

```diff
@@ -58,6 +58,85 @@ static tf::Status ValidateInputCountMin(const tf::Node* op, size_t count) {
   }
   return tf::Status::OK();
 }
+//
+// Helper for storing ops in ng_op_map.
+// For most of the cases, op would have one output so
+// vector ng_op_map[op_name] would contain one element.
+//
+// If storing more than one output_nodes, make sure it's in
+// the same order as tensorflow would do that.
+//
+// Parameters:
+//    Builder::OpMap& ng_op_map        - The TF-to-nGraph op map.
+//    std::string op_name              - Name of the op.
+//
+//    shared_ptr<ng::Node> output_node - ng::Node to store
+//
+
+static tf::Status SaveNgOp(Builder::OpMap& ng_op_map, const std::string op_name,
+                           const shared_ptr<ng::Node> output_node) {
+  // no need to try-catch, map[key] will create vector object
+  // if not exists
+  ng_op_map[op_name].push_back(output_node);
+  return tf::Status::OK();
+}
+
+// Helper for fetching correct input node from ng_op_map.
+// Handles edge checking to make sure correct input node is
+// fetched.
+//
+// Reduces some boilerplate code (incorrect from now) like this:
+//
+//      tf::Node* tf_input;
+//      TF_RETURN_IF_ERROR(op->input_node(0, &tf_input));
+//
+//      shared_ptr<ng::Node> ng_input;
+//      try {
+//        ng_input = ng_op_map.at(tf_input->name());
+//      } catch (const std::out_of_range&) {
+//        return tf::errors::NotFound(tf_input->name(),
+//                                    " is not found in the ng_op_map");
+//      }
+//
+// Into 2 lines:
+//
+//      shared_ptr<ng::node> ng_input;
+//      TF_RETURN_IF_ERROR(ng_op_map, op, 0, ng_input)
+//
+//
+//
+// Parameters:
+//    Builder::OpMap& ng_op_map     - The TF-to-nGraph op map.
+//    tf::Node* op                  - TF op being translated.
+//    input_idx                     - index of input
+//
+//    shared_ptr<ng::Node>& result  - ng::Node reference where result
+//                                    will be written
+//
+//
+
+static tf::Status GetInputNode(const Builder::OpMap& ng_op_map, tf::Node* op,
+                               int input_idx, shared_ptr<ng::Node>& result) {
+  // input op may have resulted in more than one ng::Node (eg. Split)
+  // we need to look at Edge to check index of the input op
+  std::vector<const tf::Edge*>* edges = new std::vector<const tf::Edge*>;
```

Comment:
If we `new` here this will leak memory, right? I think this should work:

```
std::vector<const tf::Edge*> edges;
TF_RETURN_IF_ERROR(op->input_edges(&edges));
// etc.
```